android/java/org/chromium/chrome/browser/shields/BraveShieldsHandler.java

```diff
@@ -170,15 +180,59 @@ public BraveShieldsHandler(Context context) {
         }
     }
 
+    public void loadDisconnectEntityList() {
+        try {
+            JSONObject obj = new JSONObject(loadDisconnectEntityJSONFromAsset());
+            JSONObject entities = obj.getJSONObject("entities");
+            Iterator<String> keysItr = entities.keys();
+            while (keysItr.hasNext()) {
+                String key = keysItr.next();
+                Object value = entities.get(key);
+                JSONArray jsonProperties = ((JSONObject) value).getJSONArray("properties");
+                JSONArray jsonResources = ((JSONObject) value).getJSONArray("resources");
+
+                for (int i = 0; i < jsonResources.length(); i++) {
+                    // Pair<String, String> resourceCompanyName = new Pair<String,
+                    // String>(jsonResources.getString(i), key);
+                    mResourceToCompanyNameList.add(new Pair(jsonResources.getString(i), key));
+                }
+            }
+            isDisconnectEntityLoaded = true;
+
+        } catch (JSONException exception) {
+            exception.printStackTrace();
+        }
+    }
+
+    private String loadDisconnectEntityJSONFromAsset() {
```

Comment:
this is not right, we should not be doing this in java and we should not be embedding an adblock list directly in the code anywhere.